tools/kms_secrets_encryption.py

```diff
@@ -0,0 +1,551 @@
+                print("{}".format(err))
+                raise SystemExit()
+        else:
+            print("Using existing crypto key: {}\n".format(crypto_key_id))
+        
+        return crypto_key_id
+
+
+    def initialize_keyring(self, key_ring_id):
+        """A method that initializes this instance's key ring.
+
+        This initialization method is called in the constructor to
+        create a default key ring if it doesn't exist.
+        
+        Args:
+            key_ring_id (str): key ring being created
+        Returns:
+            string: the key ring used
+        """
+
+        key_rings_list = self.get_key_rings()
+
+        # Create the key ring only if it doesn't exist
+        if key_ring_id not in key_rings_list:
+            try:
+                self.create_key_ring(key_ring_id)
+                print("Created key ring: {}\n".format(key_ring_id))
+        
+            except Exception as err:
+                print("An exception occurred creating new key ring:")
+                print("{}".format(err))
+                raise SystemExit()
+        else: 
+            print("Using existing key ring: {}\n".format(key_ring_id))
+
+        return key_ring_id
+
+
+    def get_crypto_keys(self, key_ring_id):
+        """A method that retrieves a list of crypto keys associated with a key ring.
+
+        This method returns a list of all the crypto keys associated with a specific key ring.
+        
+        Args:
+            key_ring_id (str): a GCP KMS key ring
+        Returns:
+            list: a list of all the crypto keys associated with the key ring argument.
+        """
+
+        parent   = self.kms_client.key_ring_path(self.project_id, self.location, key_ring_id)
+        response = self.kms_client.list_crypto_keys(parent)
+
+        # Access the name property and split string from the right. [2] to get the string after the separator
+        # eg. name: "projects/user-terraform/locations/global/keyRings/cas_keyring/cryptoKeys/cas_key"
+        crypto_keys_list = list(map(lambda key: key.name.rpartition('/')[2], response))
+
+        return crypto_keys_list
+
+
+    def get_key_rings(self):
+        """A method that retrieves a list of key rings.
+
+        This method returns a list of all the key rings associated 
+        with the GCP service account.
+        
+        Returns:
+            list: a list of all the key rings
+        """
+
+        parent   = self.kms_client.location_path(self.project_id, self.location)
+        response = self.kms_client.list_key_rings(parent)
+
+        # Access the name property and split string from the right. [2] to get the string after the separator
+        # eg. name: "projects/user-terraform/locations/global/keyRings/cas_keyring"
+        key_rings_list = list(map(lambda key_ring: key_ring.name.rpartition('/')[2], response))
+
+        return key_rings_list
+
+
+    def read_tfvars_data(self, tfvars_file):
```

Comment:
This and the secret reading function could be 1 function since they share so much in common.